apps/studio/src/features/editing-experience/components/ComplexEditorStateDrawer.tsx

```diff
@@ -137,7 +139,18 @@ export default function ComplexEditorStateDrawer(): JSX.Element {
       )
 
       // TODO: Mark removed images/files as deleted
-      // const assetsToDelete = modifiedAssets.filter((asset) => !asset.file)
+      const assetsToDelete = modifiedAssets.filter(
+        (asset) => asset.src !== undefined,
+      )
+      await Promise.all(
+        assetsToDelete.map(({ src }) => {
+          if (!src) {
+            return
+          }
+
+          return deleteAsset({ fileKey: src })
```

Comment:
since this is a `mutateAsync` call, we should add a try catch ourselves here so we handle the error. Also applies to the above `uploadAsset` call, seems like that was missed in the PR review too.